com.unity.multiplayer.mlapi/Runtime/Messaging/MessageBatcher.cs

```diff
@@ -126,40 +137,55 @@ public void QueueItem(in RpcFrameQueueItem queueItem)
                 if (sendStream.IsEmpty)
                 {
                     sendStream.IsEmpty = false;
-                    sendStream.NetworkChannel = queueItem.NetworkChannel;
-
-                    switch (queueItem.QueueItemType)
-                    {
-                        // 8 bits are used for the message type, which is an NetworkConstants
-                        case RpcQueueContainer.QueueItemType.ServerRpc:
-                            sendStream.Writer.WriteByte(NetworkConstants.SERVER_RPC); // MessageType
-                            break;
-                        case RpcQueueContainer.QueueItemType.ClientRpc:
-                            sendStream.Writer.WriteByte(NetworkConstants.CLIENT_RPC); // MessageType
-                            break;
-                    }
+                    sendStream.NetworkChannel = item.NetworkChannel;
+                }
+                // If the item is a different channel we have to flush and change channels.
```

Comment:
Couple of things:
- In the not-so-different future we want to go to "Channels as QoS"
- sometimes we think we're using different channels, but the transport is free to map to fewer or even just one channel.  Is this a case where you'd want the transport to tell you if 2 channels aren't the same "physical" channel?

Reply:
Yes, you're right. What's the correct way to get that underlying "physical" channel information? Did a brief search and I'm not finding an obvious answer.

Reply:
Actually, I think this may need more thought...

While, yes, in logical terms, we want to batch by the physical channel because that's really where the batching matters, in practical terms, NetworkTransport.Send() expects the NetworkChannel as a parameter, not the NetworkDelivery type, and the receiving end also is able to read what channel it was sent on. So if we change the batching to work that way, then the receiving end will lose the ability to differentiate between channels with the same underlying delivery guarantees if that's ever relevant to them.

I think in order to make this change, we would need to make the NetworkChannel part of the message header for each message, and then change the transport API so that its operation isn't tied to channels, but only to the NetworkDelivery values. Until we either separate those two things, or decide to deprecate channels entirely and only provide delivery values (probably not going to happen), we can't batch messages of different channels together even if they have the same underlying NetworkDelivery value.

Reply:
Thing is, the design of MLAPI is such that we cannot actually know anything about what the transport is physically doing, and I'm wondering if this is a case that points to wanting to be able to query the transport for this.  @andrews-unity this is that area of channel work I mentioned

Reply:
So I am all for re-working channels to being more of a QoS then what they are today because as @mattwalsh-unity stated all transports generally just map them to some sort of delivery that makes sense to that transport which is not consistent at all across transports. 

With that said its not clear to me what you would actually query the transport for? would we pass in a channel and then have the Transport return back the delivery method it was going to use?

I personally would love to get rid of all the MLAPI_CHANNEL stuff in the transport API re-design and move it somewhere else. I am very much liking the idea @ShadauxCat presented which is more of a soft deprecation which is we still have channel but they mean something only to the data consumer above but not to the actual transport, thus moving the channel concept to the header and only providing an API to send that pushes a delivery method vs an arbitrary byte that could mean anything to any transport.